plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -221,25 +221,68 @@ internal fun mapSnippetFindings(
                     else -> false
                 }
 
-                val isLocationsWithFalsePositives = locationsWithFalsePositives.any {
-                    it.given.sourceLocation == sourceLocation
+                val isLocationWithFalsePositives = remainingSnippetChoices.removeIf {
+                    it.given.sourceLocation == sourceLocation &&
+                        it.choice.reason == SnippetChoiceReason.NO_RELEVANT_FINDING
                 }
-
-                if (isLocationsWithFalsePositives) {
+                if (isLocationWithFalsePositives) {
                     logger.info {
                         "Ignoring snippet $purl for file ${sourceLocation.prettyPrint()}, " +
                             "as this is a location with only false positives."
                     }
                 }
 
-                if (!isSnippetChoice && !isLocationsWithFalsePositives) {
+                if (!isSnippetChoice && !isLocationWithFalsePositives) {
                     findings.getOrPut(sourceLocation) { mutableSetOf(ortSnippet) } += ortSnippet
                 }
+
+                getLicenseFindingFromSnippetChoice(
+                    remainingSnippetChoices,
+                    sourceLocation,
+                    ortSnippet
+                )?.let { finding -> snippetLicenseFindings += finding }
             }
         }
 
         findings.map { SnippetFinding(it.key, it.value) }
-    }.toSet()
+    }.toSet().also {
+        remainingSnippetChoices.forEach {
+            logger.warn {
+                "Snippet choice's snippet ${it.choice.purl} for ${it.given.sourceLocation.prettyPrint()} " +
+                    "is not in the snippet results"
```

Comment:
Missing dot at the end. Also, assign the string to a variable instead of repeating it below.
I find that sentence hard to undestand, being more verbose could help, e.g.:
"The configuration contains a snippet choice for the snippet ${it.choice.purl} at ${it.given.sourceLocation.prettyPrint()}, but the FossID result contains no such snippet."